source/1_engine/rex_engine/src/event_system.cpp

```diff
@@ -14,18 +14,75 @@ namespace rex
 {
   namespace event_system
   {
-    rsl::unordered_map<EventType, rsl::vector<EventFunction>>& delegates_map()
+    namespace internal
     {
-      static rsl::unordered_map<EventType, rsl::vector<EventFunction>> delegates_map;
-      return delegates_map;
+      class EventQueue
+      {
+      public:
+        EventQueue()
+            :m_subscriptions()
+            ,m_front()
+            ,m_back()
+            ,m_current(&m_front)
+            ,m_next(&m_back)
+        {
+        }
+
+        void subscribe(EventType type, const EventFunction& function)
+        {
+          m_subscriptions[type].push_back(function);
```

Comment:
just to note, this is not thread save, as in, the entire class is not thread safe

Reply:
I think there are a lot more functionalities within REX that are not thread-safe. As we are single-threaded for now I think it's fine.